tests/sentry/lang/native/test_sources.py

```diff
@@ -0,0 +1,161 @@
+from typing import Optional
+
+import pytest
+from django.utils import timezone
+from freezegun import freeze_time
+
+from sentry.debug_files.artifact_bundle_indexing import FlatFileIdentifier, FlatFileMeta
+from sentry.lang.native.sources import get_bundle_index_urls
+from sentry.models import ArtifactBundleFlatFileIndex
+from sentry.testutils.helpers import override_options
+
+
+def _mock_flat_file_index(
+    project_id: int, release: Optional[str], dist: Optional[str]
+) -> ArtifactBundleFlatFileIndex:
+    index = ArtifactBundleFlatFileIndex.objects.create(
+        project_id=project_id,
+        release_name=release or "",
+        dist_name=dist or "",
+        flat_file_index=None,
+    )
+    index.update_flat_file_index(file_contents="{}")
+
+    return index
+
+
+@pytest.mark.django_db
+@freeze_time("2023-07-26T10:00:00")
+@override_options({"symbolicator.sourcemaps-bundle-index-sample-rate": 0.0})
+def test_get_bundle_index_urls_with_no_zero_sample_rate(default_project):
+    release = "1.0"
+    dist = "android"
+
+    # We want to create both indexes to show that still we get no data back.
+    _mock_flat_file_index(project_id=default_project.id, release=release, dist=dist)
+    _mock_flat_file_index(project_id=default_project.id, release=None, dist=None)
+
+    debug_id_index, url_index = get_bundle_index_urls(
+        project=default_project, release=release, dist=dist
+    )
+
+    assert debug_id_index is None
+    assert url_index is None
+
+
+@pytest.mark.django_db
+@freeze_time("2023-07-26T10:00:00")
+@override_options({"symbolicator.sourcemaps-bundle-index-sample-rate": 1.0})
+def test_get_bundle_index_urls_with_no_cached_values(default_project):
+    release = "1.0"
+    dist = "android"
+
+    # We test with no data.
+    debug_id_index, url_index = get_bundle_index_urls(
+        project=default_project, release=release, dist=dist
+    )
+
+    assert debug_id_index is None
+    assert url_index is None
+    assert (
+        FlatFileIdentifier(
+            project_id=default_project.id, release=release, dist=dist
+        ).get_flat_file_meta_from_cache()
+        is None
+    )
+
+    index_1 = _mock_flat_file_index(project_id=default_project.id, release=release, dist=dist)
+
+    # We test the generation with release only.
+    debug_id_index, url_index = get_bundle_index_urls(
+        project=default_project, release=release, dist=dist
+    )
+
+    assert debug_id_index is None
+    assert (
+        url_index
+        == f"http://testserver/api/0/projects/baz/bar/artifact-lookup/?download=bundle_index/{index_1.id}/1690365600000.0"
+    )
```

Comment:
```suggestion
    assert url_index.endswith(f"artifact-lookup/?download=bundle_index/{index_1.id}/1690365600000.0"
```

Also, is the timestamp in milliseconds? if so, we probably don’t need the fraction, better to cast that to `int`.

… and the same for the other similar asserts.